### DISCUSSION TITLE
Database updates

### DISCUSSION THREAD

**Discussion by sanketgarade** [mukta-strot/marathi-shabd#34] [General]
hi @zarbod @the-kaustubh 

there are some updates and requests from my end. 
1. let us keep `develop` branch as the development branch and merge it with `main` branch when something important needs to go onto the website. So going forward please raise PRs into the `develop` branch and not `main`.
2. I am adding some words in the database and I think that the database needs some update in its content. like -
  - context is needed in case of some words, especially when the meaning of the word differs with context.
  - english sentence example may not be necessary, since we are not trying to replace entire sentences from en to mr.
  - instead we can have one or multiple marathi sentences as example (depending on the word)
  - we could also have a column for "remarks" where some additional information can be mentioned about the word.
  - there could also be some more modifications in this as the understanding gets more matured. 
  - You guys also give your suggestions, please.

So for the database, right now I'll keep the format unchanged and we can complete the markdown file scripts which we have already planned. 

Also since we are planning to have a new database in a more efficient format (json or something) as @the-kaustubh has suggested, we will have to incorporate these requirements there also. And the markdown generator scripts will also need to be later updated to read from the new format. Thats why, as much as possible, when developing the **markdown scripts** and **search site code** we can try to keep the program structure generic and reusable so that the rework is less when 1. we change from csv to the new format and 2. also flexible to accommodate the changes in the database structure as it become more mature.

@the-kaustubh have you decided what will the format of the new database file will be?

**Comment by the-kaustubh** [comment]
You can change the format of db.csv, it will only break the search site for now, but that's fixable.
Also the new db will be totally on Mongo Atlas and not in any repo.

**Reply by sanketgarade** [reply]
ok. I will update the format sometime today or tomorrow.

btw the db.csv is now updated to more than 100 entries. so we will have to migrate the data from csv to mongo through some automation. either mongo should have an option to import csv. or we need to change it into some compatible format for mongo using some script.

**Reply by sanketgarade** [reply]
database format is updated. please check. also suggest improvements if any. @the-kaustubh @zarbod
